jobs/build/appregistry/Jenkinsfile

```diff
@@ -113,37 +119,126 @@ node {
+            timeout: 60,
+            validResponseCodes: "200:599",
+        )
+    }
+
+    def getMetadataNVRs = { operatorNVRs, stream ->
+        def nvrFlags = operatorNVRs.collect { "--nvr ${it.nvr}" }.join(" ")
+        doozer("operator-metadata:latest-build --stream ${stream} ${nvrFlags}")
+    }
+
+    def attachToAdvisory = { advisory, metadata_nvrs ->
+        def elliott_build_flags = []
+        metadata_nvrs.split().each { nvr -> elliott_build_flags.add("--build ${nvr}") }
+
+        elliott """
+            find-builds -k image
+            ${elliott_build_flags.join(" ")}
+            --attach ${advisory}
+        """
+    }
+
     try {
         def operatorData = []
         sshagent(["openshift-bot"]) {
-            stage("fetch appregistry images") {
-                def include = params.IMAGES.trim()
-                if (include) {
-                    include = "--images " + commonlib.cleanCommaList(include)
+            stage("validate params") {
+                if (!validate(params)) {
+                    error "Parameter validation failed"
                 }
-                def lines = buildlib.doozer """
-                    --working-dir ${workDir}
-                    --group 'openshift-${params.BUILD_VERSION}'
-                    ${include}
-                    images:print
-                    --label 'com.redhat.delivery.appregistry'
-                    --short '{label},{name},{build},{version}'
-                """, [capture: true]
+            }
+            stage("fetch appregistry images") {
+                def lines = getImagesData(params.IMAGES.trim())
                 operatorData = processImages(lines)
+
+                if (params.STREAM in ["stage", "prod"]) {
+                    def advisoriesNVRs = fetchNVRsFromAdvisories(params.OLM_OPERATOR_ADVISORIES)
+                    operatorData = replaceNVRsByFoundInAdvisories(operatorData, advisoriesNVRs)
+
+                    if (operatorData.any { it.nvr == null }) {
+                        currentBuild.description += """\n
+                        Advisories missing operators ${operatorData.findAll { it.nvr }.collect { it.name }.join(",")}
+                        """
+                        echo """
+                        ERROR: The following operators were not found in provided advisories.
+                        ${operatorData.findAll { !it.nvr }.collect { it.name }.join(",")}
+
+                        Possible solutions:
+                        1. Add more advisories in OLM_OPERATOR_ADVISORIES parameter, that have the missing operators attached
+                        2. Attach missing operators to at least one of the provided advisories: ${params.OLM_OPERATOR_ADVISORIES}
+                        3. Limit the expected operators in IMAGES parameter: ${operatorData.findAll { it.nvr }.collect { it.name }.join(",")}
+                        """
+                        sh "exit 1"
```

Comment:
It works, but it feels like an abuse of `sh`. How about
```suggestion
                        error "operators not found"
```

Reply:
https://github.com/openshift/aos-cd-jobs/pull/2001/commits/fd4a8d7a47f4ab8f48002c716abd322010da6de3